cmd/kubeadm/app/cmd/init.go

```diff
@@ -342,20 +362,41 @@ func newInitData(cmd *cobra.Command, args []string, options *initOptions, out io
 		}
 	}
 
+	var certificateKey []byte
+	if options.uploadCerts {
+		certificateKey, err = cryptoutil.CreateRandBytes(kubeadmconstants.CertificateKeySize)
```

Comment:
During the initial planning we had discussed allowing a user to specify an encryption key, is that work being tracked elsewhere?

Reply:
config changes were definitely punted for a future release, but in terms of CLI we might allow this in this cycle. we can do this in a separate PR if there is time.

@fabriziopandini WDYT?
we don't seem to have tracking for that custom flag here (but i might have missed it):
https://github.com/kubernetes/kubeadm/issues/1373
so perhaps you had this in mind for a next cycle when we also add the config changes?

Reply:
allowing a user to specify an encryption key is liked to config changes (punted).
However, I understand that this is a kind of blocker for higher level tools, so let's discuss how to address this next kubeadm office hours

Reply:
We decided in kubeadm meeting to add the flag  `--certificate-key`. https://github.com/kubernetes/kubeadm/issues/1408